hawc/apps/assessment/models.py

```diff
@@ -737,10 +737,64 @@ def try_read_df(cls, data, suffix: str, worksheet_name: str = None) -> pd.DataFr
         return df
 
 
+class Job(models.Model):
+    PENDING = 1
+    SUCCESS = 2
+    FAILURE = 3
+    STATUS_CHOICES = (
+        (1, "PENDING"),
+        (2, "SUCCESS"),
+        (3, "FAILURE"),
+    )
+
+    TEST = 1
+    JOB_CHOICES = ((1, "TEST"),)
+
+    def test(self, fail=False):
+        if fail:
+            raise Exception("FAILURE")
+        return "SUCCESS"
+
+    task_id = models.UUIDField(primary_key=True, editable=False)
+    assessment = models.ForeignKey(
+        Assessment, null=True, blank=True, on_delete=models.CASCADE, related_name="jobs"
+    )
+    status = models.PositiveSmallIntegerField(
+        choices=STATUS_CHOICES, default=PENDING, editable=False
+    )
+    job = models.PositiveSmallIntegerField(choices=JOB_CHOICES, default=TEST)
+
+    kwargs = JSONField(default=dict)
+    result = models.TextField(blank=True, editable=False)
+    exception = models.TextField(blank=True, editable=False)
+
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        ordering = ("-created",)
+
+    def set_exception(self, exception):
+        self.exception = exception
+        self.status = self.FAILURE
+
+    def set_result(self, result):
+        self.result = result
+        self.status = self.SUCCESS
+
+    def execute(self):
```

Comment:
i think the post-save signal should `run_job.apply_async(task_id=self.task_id)`, and then the execute method here can essentially do the get_func behavior.. We should also indicate that execute is expected to return something, I'd probably say it should return a `Dict` since that's we'll save the result field